osquery/tables/networking/windows/interfaces.h

```diff
@@ -0,0 +1,20 @@
+/*
+*  Copyright (c) 2014-present, Facebook, Inc.
+*  All rights reserved.
+*
+*  This source code is licensed under the BSD-style license found in the
+*  LICENSE file in the root directory of this source tree. An additional grant
+*  of patent rights can be found in the PATENTS file in the same directory.
+*
+*/
+
+#pragma once
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+namespace osquery {
+namespace tables {
+QueryData genInterfaceDetails(QueryContext& context);
```

Comment:
This is directly used in the `interface_details` table, but I added the header to make the data also available to the `arp_cache` query so I can match up the interface to a mac address for display. 

This is because the raw output from the `MSFT_NetNeighbor` class lacks that data, e.g.
```shell
PS C:\> Get-CimInstance msft_netneighbor -Namespace root/StandardCimv2 | select -first 1


Caption                    :
Description                :
ElementName                :
InstanceID                 :
CommunicationStatus        :
DetailedStatus             :
HealthState                :
InstallDate                :
Name                       : pp:<DD;Dpp=lDloBC55B55;
OperatingStatus            :
OperationalStatus          :
PrimaryStatus              :
Status                     :
StatusDescriptions         :
AvailableRequestedStates   :
EnabledDefault             : 2
EnabledState               : 5
OtherEnabledState          :
RequestedState             : 12
TimeOfLastStateChange      :
TransitioningToState       : 12
CreationClassName          :
SystemCreationClassName    :
SystemName                 :
AccessContext              : 0
AccessInfo                 :
InfoFormat                 :
OtherAccessContext         :
OtherInfoFormatDescription :
AddressFamily              : 23
InterfaceAlias             : Ethernet 2
InterfaceIndex             : 8
IPAddress                  : ff02::1:ff3b:be89
LinkLayerAddress           : 33-33-FF-3B-BE-89
State                      : 6
Store                      : 1
PSComputerName             :
```

As you can see the only data we get are the `InterfaceAlias` or the `InterfaceIndex` fields, so we have to pull the data from another source to match up the data with what is output in `arp -a`